liquidctl/driver/kraken3.py

```diff
@@ -392,17 +424,235 @@ def _write_colors(self, cid, mode, colors, sval, direction):
             footer = [backward_byte, color_count, mode_related, static_byte, led_size]
             self._write(header + color + footer)
 
+    def set_screen(self, channel, mode, value, **kwargs):
+        """Not supported by this device."""
+        raise NotSupportedByDevice()
+
 
 class KrakenZ3(KrakenX3):
     """Fourth-generation Kraken Z liquid cooler."""
 
     SUPPORTED_DEVICES = [
-        (0x1e71, 0x3008, None, 'NZXT Kraken Z (Z53, Z63 or Z73)', {
-            'speed_channels': _SPEED_CHANNELS_KRAKENZ,
-            'color_channels': _COLOR_CHANNELS_KRAKENZ,
-        })
+        (
+            0x1E71,
+            0x3008,
+            None,
+            "NZXT Kraken Z (Z53, Z63 or Z73)",
+            {
+                "speed_channels": _SPEED_CHANNELS_KRAKENZ,
+                "color_channels": _COLOR_CHANNELS_KRAKENZ,
+            },
+        )
     ]
 
+    def __init__(self, device, description, speed_channels, color_channels, **kwargs):
+        super().__init__(device, description, speed_channels, color_channels, **kwargs)
+
+        if sys.platform in ["win32", "cygwin"]:
+            self.bulk_device = WinUsbPy()
+            found_device = self._find_winusb_device(
+                "vid_1e71", "pid_3008", self.device.serial_number
+            )
+            assert found_device, "Cannot find bulk out device"
+        else:
+            self.bulk_device = next(
+                (
+                    handle
+                    for handle in PyUsbDevice.enumerate(self.vendor_id, self.product_id)
+                    if handle.serial_number == self.device.serial_number
+                ),
+                None,
+            )
+
+            assert self.bulk_device, "Cannot find bulk out device"
+
+            self.bulk_device.open()
+
+        self.orientation = 0  # 0 = Normal, 1 = +90 degrees, 2 = 180 degrees, 3 = -90(270) degrees
+        self.brightness = 50  # default 50%
+
+    def _find_winusb_device(self, vid, pid, serial):
```

Comment:
I know very little about WinUSB, but your implementation does make sense to me.

---

Excuse me if I missed something, but wouldn't this be a relatively smaller and simpler change if made directly to WinUsbCDC?  It also seems to me like something others would be interested in...